apps/ledger-live-desktop/src/renderer/families/evm/nft.ts

```diff
@@ -13,13 +13,23 @@ export const injectNftIntoTransaction = (
   transaction: EvmTransaction,
   nftProperties: Partial<NftProperties>,
   standard?: NFTStandard,
-): EvmTransaction => ({
-  ...transaction,
-  mode: (standard ? standard.toLowerCase() : transaction.mode) as Lowercase<NFTStandard>,
-  nft: {
-    collectionName: transaction.nft?.collectionName ?? "",
-    contract: nftProperties?.contract ?? transaction.nft?.contract ?? "",
-    tokenId: nftProperties?.tokenId ?? transaction.nft?.tokenId ?? "",
-    quantity: nftProperties?.quantity ?? transaction.nft?.quantity ?? new BigNumber(NaN),
-  },
-});
+): EvmTransaction => {
+  if (transaction.mode === "send") {
+    throw new Error("Expected an NFT transaction but received a Send mode transactions");
+  }
+
+  if (standard && standard === "SPL") {
+    throw new Error("Expected an ERC721 or ERC1155 standard but received an SPL standard");
+  }
+
+  return {
+    ...transaction,
+    mode: standard ? <Lowercase<typeof standard>>standard.toLowerCase() : transaction.mode,
```

Comment:
Here I'm doing the checks to narrow the type correctly and to expose any bad configuration to the dev when testing the code

I'm also only casting the type for toLowerCase now instead of casting mode too and it exposes the fact the we could receive a transaction in send mode

It was using as before because adding an nft field on the transaction removes the send option from the union of mode and make transction.mode incompatible if you don't guard for it first